extensions/arc/runtime/src/main/java/io/quarkus/arc/runtime/ConfigMappingCreator.java

```diff
@@ -23,6 +24,10 @@ public Object create(SyntheticCreationalContext<Object> context) {
             throw new IllegalStateException("No current injection point found");
         }
 
+        if ((boolean) context.getParams().get("nativeBuild")) {
+            NativeBuildConfigCheckInterceptor.verifyCurrentImageMode(Set.of());
```

Comment:
IIRC the Set is empty here because we assume none of the properties being injected are "build and runtime fixed"... but can't a config mapping be completely "build and runtime fixed"? If so we'd want to skip this check, no?

Reply:
Yes, a `ConfigRoot` defined as a config mapping is something I need to check.

Because we do register a synthetic bean for any `BUILD_AND_RUN_TIME_FIXED`/`RUN_TIME` config root [here](https://github.com/quarkusio/quarkus/blob/main/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java#L238-L257)  and my guess would be that an injected config root with a config mapping would result in an ambiguous dependency error...

Reply:
FYI it's not a problem because a config root implemented as a config mapping is not considered a `RootDefinition`...